### PR TITLE
Use system's share UI when sharing a session as text

### PR DESCRIPTION
**Pull request by cketti**
# Description
- Use system's share UI for "Share as text"
- Doesn't change the behavior for "Share as JSON (for Chaosflix)" so Chaosflix can be selected as default app
- `SessionSharer` was converted to Kotlin

# Before
EventFahrplan_share_as_text_old

# After
EventFahrplan_share_as_text_new

### CODE REVIEW

**Comment by johnjohndoe** [comment]
@cketti Thanks for this modernization. Please document on which Android version / devices you tested this.

**Comment by cketti** [comment]
Tested on a Pixel 2 running Android 11 beta.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+object SessionSharer {
+    // String formattedSessions can be one or multiple sessions
+    @JvmStatic
+    fun shareSimple(context: Context, formattedSessions: String) {
+        val intent = Intent(Intent.ACTION_SEND).apply {
+            type = "text/plain"
+            putExtra(Intent.EXTRA_TEXT, formattedSessions)
+        }
+
+        // Show system's share UI. It handles the case of no matching apps.
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.kt:16]: What happens exactly when no app is found?

Comment (cketti) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.kt:16]: Depends on the Android version. Various ways of displaying a "No suitable apps found" message. On Android 11 it's this:

EventFahrplan_share_fail


You can easily force this by using a made up intent action value.

Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.kt:16]: OK. Thanks for the clarification.

**Comment by johnjohndoe** [comment]
# Successfully tested on
with `ccc36c3` flavor, `debug` build
- :heavy_check_mark: Nexus 5X emulator, Android 4.1.2 (API 16)
- :heavy_check_mark: Nexus 5 emulator, Android 4.3.1 (API 18)

On both emulators sharing a session directly opens the _Messaging_ app without chooser UI. This might result from the fact the emulator has no alternative app installed.

Messaging app screenshot

**Review by johnjohndoe — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**